<?xml version="2.0" ?>
<launch>
    <!-- This launch file is specified for the bagfile mir_test.bag -->
    
   <node pkg= "rosbag" type="play" name="player" output="screen" args="--clock --pause   /home/matchos/catkin_ws/src/laser_based_localization/bagfile/gazebo_test2.bag"/> 


   <node pkg="laser_based_localization" type="laserfusion.py" name="scan_fusion" >
   


   <!-- Here define the topic name of Laserscan -->
   <param  name="f_scan"  value="/master/f_scan" />
   <param name = "b_scan"  value ="/master/b_scan" />

   <!--Here define frame id     -->
   <param name = "front_link" value="master/front_laser_link" />
   <param name = "back_link" value="master/back_laser_link" />
   <param name = "map" value="map" />
   </node>

   <node pkg="laser_based_localization" type="localization.py" name="laser_localization">
   
   <param name = "map_path" value="/home/matchos/catkin_ws/src/laser_based_localization/map/gazebo_map.txt" />

   <param name = "map_link" value="map" />

   <!--  define the number of robots excepts master -->
   <param name = "number_of_robots" value="1" />

   <!--  master robot position -->
   <param name = "master_initial_x" value="0" />
   <param name = "master_initial_y" value="0" />
   <param name = "master_odometry"  value="/master/odometry/filtered" />



   <!-- one robot situation, define initial position and topic name and frame_id -->

   <param name = "robot1_initial_x" value="2.5" />
   <param name = "robot1_initial_y" value="2.5" />
   <param name = "robot1_odometry_link"  value="robot1/reference" />
   <param name = "robot1_odometry"  value="/robot1/odometry/filtered" />

   </node>


 

</launch>

